src/freon.c

```diff
@@ -152,24 +154,44 @@ int ioctl(int fd, unsigned long int request, ...) {
     return ret;
 }
 
+static int _open(int (*origfunc)(const char *pathname, int flags, mode_t mode),
+                 const char *origname, const char *pathname, int flags, mode_t mode);
+
 int open(const char *pathname, int flags, ...) {
     if (!orig_open) preload_init();
 
     va_list argp;
```

Comment:
I tried this, but two issues came up.  First, when trying to call va_start in _open, gcc complains:

> error: ‘va_start’ used in function with fixed args

So I had to modify _open's function signature to make it variadic, which is a bit misleading, but [here's the result](https://github.com/mkasick/crouton/commit/freon-test).

The second problem is that the va_start macro doesn't merely use the reference operator on the last argument, but instead calls __builtin_va_start passing the argument "directly".  Whatever built-in magic happens there doesn't obtain the right reference, so mode isn't fetched properly.

Thus, I'm not sure how to keep the va-args stuff in _open aside from passing a va_list, which also requires calling va_start in the separate functions.  Thoughts?